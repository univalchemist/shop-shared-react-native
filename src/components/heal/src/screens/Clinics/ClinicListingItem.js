import React from 'react';
import PropTypes from 'prop-types';
import { Dimensions, TouchableOpacity } from 'react-native';
import {
  Box,
  Text,
  PlainText,
  ListItemWithRightChevron,
  Image,
} from '@heal/src/wrappers/components';
import { connect } from 'react-redux';
import { useTheme, useIntl } from '@heal/src/wrappers/core/hooks';

import { iconLocation, panelDoctor } from '@components/heal/images';

const screenSize = Dimensions.get('window');

const ClinicListingItem = ({ clinic, latitude, longitude, onPress }) => {
  const intl = useIntl();
  const theme = useTheme();

  const shortenDistance = distance => {
    if (distance > 1000) {
      return intl.formatMessage(
        { id: 'heal.clinic.kmdistance' },
        {
          number: distance && (distance / 1000).toFixed(2),
        },
      );
    }

    return intl.formatMessage(
      { id: 'heal.clinic.mdistance' },
      {
        number: distance && distance.toFixed(2),
      },
    );
  };

  return (
    <TouchableOpacity onPress={onPress}>
      <ListItemWithRightChevron
        width={screenSize.width}
        height={134}
        justifyContent="center"
        paddingLeft={32}
        paddingRight={30}
      >
        <Text
          fontSize={21}
          color={theme.colors.defaultText}
          lineHeight={26}
          letterSpacing={0.3}
        >
          {clinic.name}
        </Text>
        <PlainText numberOfLines={1}>{clinic.address}</PlainText>
        <Box flexDirection="row" marginTop={16}>
          <Box flexDirection="row">
            <Image source={panelDoctor} />
            <Text
              fontSize={14}
              lineHeight={18}
              letterSpacing={0.3}
              marginLeft={8}
            >
              {clinic.doctors.length === 1
                ? intl.formatMessage(
                    { id: 'clinic.singularDoctor' },
                    { number: clinic.doctors.length },
                  )
                : intl.formatMessage(
                    { id: 'clinic.pluralDoctor' },
                    { number: clinic.doctors.length },
                  )}
            </Text>
          </Box>
          {latitude === 0 ||
          longitude === 0 ||
          clinic.distanceToClient === undefined ? null : (
            <Box flexDirection="row" marginLeft={32} marginRight={32}>
              <Image source={iconLocation} />
              <Text
                fontSize={14}
                lineHeight={18}
                letterSpacing={0.3}
                marginLeft={8}
              >
                {shortenDistance(clinic.distanceToClient)}
              </Text>
            </Box>
          )}
        </Box>
      </ListItemWithRightChevron>
    </TouchableOpacity>
  );
};

ClinicListingItem.propTypes = {
  clinic: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    doctors: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        locale: PropTypes.string,
        name: PropTypes.string,
        gender: PropTypes.string,
        isActive: PropTypes.bool,
      }),
    ),
    address: PropTypes.string,
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    distanceToClient: PropTypes.number,
    clinicProviderId: PropTypes.number,
  }).isRequired,
};

const mapStateToProps = ({
  heal: {
    location: { latitude, longitude },
  },
}) => ({
  latitude,
  longitude,
});

export default connect(mapStateToProps, {})(ClinicListingItem);
