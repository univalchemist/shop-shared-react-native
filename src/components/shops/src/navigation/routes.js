export const NAVIGATION_HOME_SCREEN_PATH = 'SHOP_HomeScreen';

export const NAVIGATION_HOME_PRODUCT_PATH = 'SHOP_Home_Product';

export const NAVIGATION_HOME_STACK_PATH = 'SHOP_Home';

export const SHOP_SELECT_CATEGORY_MODAL = 'SHOP_SELECT_CATEGORY_MODAL';
export const SHOP_SORT_MODAL = 'SHOP_SORT_SCREEN';
export const SHOP_FILTER_MODAL = 'SHOP_FILTER_SCREEN';

export const SHOP_CART_SCREEN = 'SHOP_CART_SCREEN';
export const SHOP_EDIT_PRODUCT_SCREEN = 'SHOP_EDIT_PRODUCT_SCREEN';
export const SHOP_REVIEW_SCREEN = 'SHOP_REVIEW_SCREEN';
export const SHOP_CHECKOUT_SCREEN = 'SHOP_CHECKOUT_SCREEN';

export const SHOP_DELIVERY_ADDRESSES_SCREEN = 'SHOP_DELIVERY_ADDRESSES_SCREEN';
export const SHOP_ADD_ADDRESS_SCREEN = 'SHOP_ADD_ADDRESS_SCREEN';
export const SHOP_EDIT_ADDRESS_SCREEN = 'SHOP_EDIT_ADDRESS_SCREEN';
export const SHOP_SINGLE_SELECT_MODAL = 'SHOP_SINGLE_SELECT_MODAL';

export const ORDER_HISTORY_SCREEN = 'ORDER_HISTORY_SCREEN';
export const ORDER_HISTORY_DETAIL_SCREEN = 'ORDER_HISTORY_DETAIL_SCREEN';
export const TRACK_ORDER_SCREEN = 'TRACK_ORDER_SCREEN';
export const WRITE_REVIEW_SCREEN = 'WRITE_REVIEW_SCREEN';
export const REVIEW_SUBMITTED_SCREEN = 'REVIEW_SUBMITTED_SCREEN';
