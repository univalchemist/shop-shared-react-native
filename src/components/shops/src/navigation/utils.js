export const getDefaultHeader = theme => {
  return {
    headerStyle: {
      backgroundColor: theme.colors.primary,
    },
    headerTitleStyle: {
      ...theme.typography.titleTextSemiBold,
      alignSelf: 'center',
    },
    headerBackTitle: null,
    headerTintColor: theme.colors.appbarTint,
  };
};
