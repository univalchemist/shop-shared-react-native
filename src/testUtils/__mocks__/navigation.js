const subscriptionMap = new Map();
const navigation = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  goBack: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  reset: jest.fn(),
  setOptions: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  addListener: (state, callback) => {
    const subscriptions = subscriptionMap.get(state) || [];
    subscriptions.push(callback);
    subscriptionMap.set(state, subscriptions);
    return {
      remove: () => {
        const subscriptions = subscriptionMap.get(state);
        subscriptionMap.set(subscriptions.filter(x => x !== callback));
      },
    };
  },

  mockEmitState: state => {
    const subscriptions = subscriptionMap.get(state) || [];
    subscriptions.forEach(x => x());
  },

  clearAll: () => {
    subscriptionMap.clear();
  },
};

export default navigation;
