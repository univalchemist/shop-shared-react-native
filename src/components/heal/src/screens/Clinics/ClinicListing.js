import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useTheme, useIntl } from '@heal/src/wrappers/core/hooks';
import { FlatList, StyleSheet, Dimensions } from 'react-native';
import { Box, Text, ErrorPanel } from '@heal/src/wrappers/components';
import ClinicListingItem from './ClinicListingItem';
import { setDetailsClinic } from '@heal/src/store/actions';
import { CLINIC_DETAILS } from '@routes';
import { SpinningLoader } from '@heal/src/components';

const screenSize = Dimensions.get('window');

const styles = StyleSheet.create({
  clinicGroups: {
    flex: 1,
  },
});

const Header = ({ total }) => {
  const intl = useIntl();
  const theme = useTheme();

  return (
    <Box
      marginTop={24}
      width={screenSize.width}
      height={38}
      justifyContent="center"
    >
      <Text
        color={theme.colors.gray[0]}
        paddingLeft={32}
        paddingRight={32}
        fontWeight="bold"
      >
        {total === 1
          ? intl.formatMessage(
              { id: 'clinic.singularFound' },
              { number: total },
            )
          : intl.formatMessage({ id: 'clinic.pluralFound' }, { number: total })}
      </Text>
    </Box>
  );
};

const ClinicListing = ({
  getClinics,
  setDetailsClinic,
  navigation,
  clinics,
  page,
  total,
  isLoading,
}) => {
  const intl = useIntl();
  const theme = useTheme();

  if (isLoading && clinics.length === 0) return <SpinningLoader />;

  if (clinics && clinics.length > 0) {
    return (
      <FlatList
        scrollIndicatorInsets={{ right: 1 }}
        style={styles.clinicGroups}
        data={clinics}
        ListHeaderComponent={() => {
          return <Header total={total} />;
        }}
        renderItem={({ item, index }) => {
          return (
            <ClinicListingItem
              clinic={item}
              onPress={() => {
                setDetailsClinic(item);
                navigation.navigate(CLINIC_DETAILS, {
                  clinic: item,
                });
              }}
            />
          );
        }}
        keyExtractor={(item, index) => index.toString()}
        onEndReachedThreshold={0.5}
        onEndReached={() => getClinics()}
      />
    );
  }

  return (
    <Box flex={1}>
      <ErrorPanel
        heading={intl.formatMessage({ id: 'clinic.noClinics' })}
        description={intl.formatMessage({ id: 'clinic.noClinicsDescription' })}
      />
    </Box>
  );
};

ClinicListing.propTypes = {
  clinics: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      doctors: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.number,
          locale: PropTypes.string,
          name: PropTypes.string,
          gender: PropTypes.string,
          isActive: PropTypes.bool,
        }),
      ),
      address: PropTypes.string,
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      distanceToClient: PropTypes.number,
      clinicProviderId: PropTypes.number,
    }),
  ).isRequired,
};

const mapStateToProps = ({
  heal: {
    clinicData: { clinics, page, total },
  },
}) => ({
  clinics,
  page,
  total,
});

export default connect(mapStateToProps, { setDetailsClinic })(ClinicListing);
