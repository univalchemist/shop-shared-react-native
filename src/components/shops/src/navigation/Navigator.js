import React from 'react';
import {
  createStackNavigator,
  TransitionPresets,
} from '@react-navigation/stack';
import { connect } from 'react-redux';
import { useIntl } from '@shops/wrappers/core/hooks';
import {
  HomeScreen,
  SelectCategoryScreen,
  FilterByScreen,
  SortByScreen,
  ProductDetailsScreen,
  ReviewScreen,
} from '@shops/screens/home';
import {
  AddAddressScreen,
  DeliveryAddressesScreen,
  EditAddressScreen,
} from '@shops/screens/deliveryAddress';
import { ModalBackButton } from '@shops/wrappers/components';
import { ShopHeader, ClearAllButton } from '@shops/components';
import {
  CartScreen,
  EditProductScreen,
  CheckoutScreen,
} from '@shops/screens/cart';
import { updateSortType, updateFilterTypes } from '@shops/store/actions';

import * as routes from './routes';
import {
  OrderHistoryDetailScreen,
  OrderHistoryScreen,
  ReviewSubmittedScreen,
  TrackOrderScreen,
  WriteReviewScreen,
} from '@shops/screens/orderhistory';
import { SingleSelectModal } from '@wrappers/components/form';

const Stack = createStackNavigator();

const Navigator = ({ updateSortType, updateFilterTypes }) => {
  const intl = useIntl();

  const onClearAllSortByPress = () => {
    updateSortType(null);
  };

  const onClearAllFilterByPress = () => {
    updateFilterTypes({});
  };

  return (
    <Stack.Navigator
      initialRouteName={routes.NAVIGATION_HOME_SCREEN_PATH}
      screenOptions={{ headerBackTitleVisible: false }}
      headerMode="float"
    >
      <Stack.Screen
        name={routes.NAVIGATION_HOME_SCREEN_PATH}
        component={HomeScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.NAVIGATION_HOME_PRODUCT_PATH}
        component={ProductDetailsScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CART_SCREEN}
        component={CartScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CHECKOUT_SCREEN}
        component={CheckoutScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_PRODUCT_SCREEN}
        component={EditProductScreen}
        options={() => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.editProduct' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_SELECT_CATEGORY_MODAL}
        component={SelectCategoryScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.selectCategory' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_SORT_MODAL}
        component={SortByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.sortBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllSortByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_FILTER_MODAL}
        component={FilterByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.filterBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllFilterByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_REVIEW_SCREEN}
        component={ReviewScreen}
        options={({ route }) => {
          const ratingCount = route.params.product
            ? route.params.product.ratingsCount
            : 0;
          const title = intl.formatMessage(
            {
              id: 'shop.review.title',
              defaultMessage: `Review(${ratingCount})`,
            },
            { ratingCount },
          );

          return {
            title,
            headerBackImage: () => <ModalBackButton />,
            ...TransitionPresets.ModalTransition,
          };
        }}
      />
      <Stack.Screen
        name={routes.SHOP_DELIVERY_ADDRESSES_SCREEN}
        component={DeliveryAddressesScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_ADD_ADDRESS_SCREEN}
        component={AddAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.addNewAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_ADDRESS_SCREEN}
        component={EditAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.editAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.ORDER_HISTORY_SCREEN}
        component={OrderHistoryScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={OrderHistoryDetailScreen}
        name={routes.ORDER_HISTORY_DETAIL_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={TrackOrderScreen}
        name={routes.TRACK_ORDER_SCREEN}
        options={{
          title: intl.formatMessage({
            id: 'shop.orderHistory.trackOrderTitle',
          }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        component={WriteReviewScreen}
        name={routes.WRITE_REVIEW_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={ReviewSubmittedScreen}
        name={routes.REVIEW_SUBMITTED_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_SINGLE_SELECT_MODAL}
        component={SingleSelectModal}
        options={({ route }) => ({
          title: route.params?.title,
          headerBackImage: ModalBackButton,
          ...TransitionPresets.ModalTransition,
        })}
      />
    </Stack.Navigator>
  );
};

export const ShopNavigator = connect(null, {
  updateSortType,
  updateFilterTypes,
})(Navigator);
