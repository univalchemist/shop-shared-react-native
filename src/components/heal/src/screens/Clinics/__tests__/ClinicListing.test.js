import React from 'react';
import { render, renderForTest } from '@testUtils';
import navigation from '@testUtils/__mocks__/navigation';
import { flushMicrotasksQueue } from 'react-native-testing-library';

import ClinicListing from '../ClinicListing';
import ClinicListingItem from '../ClinicListingItem';

const clinics = [
  {
    id: 336,
    name: 'Clinic Bukit Rimau',
    latitude: 3.1275345,
    longitude: 101.6412337,
    distanceToClient: 11300034.6201797,
    address: 'Ho Chi Minh City, Vietnam',
    doctors: [
      {
        id: 322,
        locale: 'en-HK',
        name: 'Anna Tan',
        gender: 'Female',
        isActive: true,
        clinics: [],
      },
      {
        id: 323,
        locale: 'en-HK',
        name: 'Clinic Bukit Rimau',
        gender: 'Female',
        isActive: true,
        clinics: [],
      },
    ],
  },
  {
    id: 335,
    name: 'Clinic Jelutong Shah Alam',
    latitude: 3.1277049,
    longitude: 101.6416511,
    qrCode: '1771',
    district: 'Shah Alam',
    distanceToClient: 11300080.7479106,
    address: 'Taipei, Taiwan',
    doctors: [
      {
        id: 321,
        locale: 'en-HK',
        name: 'Clinic Jelutong Shah Alam',
        gender: 'Male',
        isActive: true,
        clinics: [],
      },
    ],
  },
  {
    id: 333,
    name: 'Clinic Subang',
    latitude: 3.1259714,
    longitude: 101.6423915,
    qrCode: '2554',
    district: 'Subang Jaya',
    distanceToClient: 11300165.1150293,
    address: 'New York, USA',
    doctors: [
      {
        id: 317,
        locale: 'en-HK',
        name: 'Shantini',
        gender: 'Female',
        isActive: true,
        clinics: [],
      },
      {
        id: 318,
        locale: 'en-HK',
        name: 'Clinic Subang',
        gender: 'Male',
        isActive: true,
        clinics: [],
      },
    ],
  },
];

const getClinics = jest
  .fn()
  .mockResolvedValue({ data: { items: clinics, totalCount: 3, page: 1 } });

describe('ClinicListing', () => {
  test('Render Clinic Listing', async () => {
    // const [screen] = render(<ClinicListing navigation={navigation} />, {
    //   initialState: {
    //     heal: { clinicData: { clinics: [], page: 0, total: 0 } },
    //   },
    //   api: { getClinics },
    // });

    // await flushMicrotasksQueue();
    // const items = screen.queryAllByType(ClinicListingItem);
    // expect(items).toHaveLength(3);

    const component = renderForTest(<ClinicListing navigation={navigation} />, {
      initialState: {
        heal: { clinicData: { clinics: [], page: 0, total: 0 } },
      },
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
