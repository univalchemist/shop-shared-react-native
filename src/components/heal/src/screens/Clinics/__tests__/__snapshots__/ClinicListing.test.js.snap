// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicListing Render Clinic Listing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        flex={1}
                        style={
                          Array [
                            Object {
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          contentContainerStyle={
                            Array [
                              Object {
                                "flex": 1,
                                "justifyContent": "center",
                              },
                              Object {},
                            ]
                          }
                        >
                          <View>
                            <View
                              alignItems="center"
                              display="flex"
                              p={4}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "display": "flex",
                                    "paddingBottom": 32,
                                    "paddingLeft": 32,
                                    "paddingRight": 32,
                                    "paddingTop": 32,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {},
                                  ]
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "testUri": "../../@cxa-rn/components/src/ErrorPanel/images/errorPanel.png",
                                    }
                                  }
                                  style={
                                    Array [
                                      Object {},
                                    ]
                                  }
                                />
                              </View>
                              <View
                                mt={4}
                                style={
                                  Array [
                                    Object {
                                      "marginTop": 32,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {},
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.0"
                                    fontFamily="default"
                                    fontSize={26}
                                    fontWeight="light"
                                    letterSpacing={-1.5}
                                    lineHeight={37}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#212121",
                                          "fontFamily": "Arial",
                                          "fontSize": 26,
                                          "fontWeight": "300",
                                          "letterSpacing": -1.5,
                                          "lineHeight": 37,
                                          "textAlign": "center",
                                        },
                                      ]
                                    }
                                    textAlign="center"
                                  >
                                    No clinics available
                                  </Text>
                                </View>
                                <View
                                  mt={3}
                                  style={
                                    Array [
                                      Object {
                                        "marginTop": 16,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "lineHeight": 22,
                                          "textAlign": "center",
                                        },
                                      ]
                                    }
                                    textAlign="center"
                                  >
                                    There are no clinics matching your search criteria.
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
